### PR TITLE
Add a convention connecting credential specs and the issued VCs

### PR DESCRIPTION
**Pull request by przydatek**
Add a convention for connecting credential specifications with the issued credentials. 
The convention enables automated validation of VC-claims in issued credentials wrt. credential spec used by a relying party.  Also, adapt the existing demo issuer to follow the convention.

A helper for the automated validation will be added to `vc_util` in a follow-up PR.

### CODE REVIEW

File: demos/vc_issuer/tests/issue_credential.rs [hunk 1]
```diff
@@ -617,6 +617,11 @@ fn should_issue_credential_e2e() -> Result<(), CallError> {
                 prepared_context: prepared_credential.prepared_context,
             },
         )?;
+
+        println!(
```
Comment (frederikrothenberger) [inline demos/vc_issuer/tests/issue_credential.rs:621]: Did you intend to leave this here?

File: src/vc_util/src/lib.rs [hunk 2]
```diff
@@ -372,8 +372,16 @@ fn validate_claims_match_spec(
     let subject = Subject::from_json_value(credential_subject.clone()).map_err(|_| {
         inconsistent_jwt_claims("missing credentialSubject in VerifiedAdult JWT vc")
     })?;
+    let credential_type = &spec.credential_type;
```
Comment (frederikrothenberger) [inline src/vc_util/src/lib.rs:375]: Shouldn't this now enable _generic_ validation? I.e. this function should not specifically validate anything related to "adult", right?

**Review by frederikrothenberger — COMMENTED** [review]
Using this convention, the vc_util custom module could be removed, right? And be replaced by a generic validation function that takes a `CredentialSpec` as input?

Is this planned in a separate PR?

**Comment by przydatek** [comment]
> Using this convention, the vc_util custom module could be removed, right? And be replaced by a generic validation function that takes a `CredentialSpec` as input?
Yes and yes.

> Is this planned in a separate PR?
Yes.

File: demos/vc_issuer/tests/issue_credential.rs (hunk 1 above)
Comment (przydatek) [reply demos/vc_issuer/tests/issue_credential.rs]: Oops, not intended. Thanks for catching, removed now.

File: src/vc_util/src/lib.rs (hunk 2 above)
Comment (przydatek) [reply src/vc_util/src/lib.rs:375]: Yes, this does enable generic validation, which will come in a follow-up PR.

**Review by przydatek — COMMENTED** [review]
Thanks, PTAL.

**Review by nmattia — APPROVED** [review]
LGTM!

**Pull request merged**